pkg/sensors/tracing/killer.go

```diff
@@ -140,32 +149,77 @@ func createKillerSensor(
 	var load *program.Program
 	var progs []*program.Program
 	var maps []*program.Map
+	specOpts, err := getSpecOptions(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get spec options: %s", err)
+	}
 
-	useMulti := !option.Config.DisableKprobeMulti && bpf.HasKprobeMulti()
-
-	attach := fmt.Sprintf("%d syscalls: %s", len(syscallsSyms), syscallsSyms)
-	prog := sensors.PathJoin(name, "killer_kprobe")
+	if !bpf.HasSignalHelper() {
+		return nil, fmt.Errorf("killer sensor requires signal helper which is not available")
+	}
 
-	if useMulti {
-		load = program.Builder(
-			path.Join(option.Config.HubbleLib, "bpf_multi_killer.o"),
-			attach,
-			"kprobe.multi/killer",
-			prog,
-			"killer")
+	// select proper override method based on configuration and spec options
+	overrideMethod := specOpts.OverrideMethod
```

Comment:
maybe this (with the switch below) could be in separate function that returns overrideMethos or error